src/workerd/jsg/jsg.h

```diff
@@ -2085,6 +2085,20 @@ class Lock {
     return *reinterpret_cast<Lock*>(v8Isolate->GetData(2));
   }
 
+  // Manually make adjustments to the amount of external memory reported to V8.
+  // This is useful when we have a large amount of external memory allocated that
+  // typically would not be visible to v8's memory tracking. In the future we hope
+  // to make most memory accounting automatic, making most direct uses of this
+  // API unnecessary but there will always be times when manual adjustments are
+  // necessary.
+  void adjustExternalMemory(ssize_t amount);
+
+  // Reports amount of external memory to be manually attributed to the isolate.
+  // When the returned kj::Own<void> is dropped, the amount will be subtracted
+  // from the isolate's external memory accounting. It is important that these
+  // be held onto only during the lifetime of the isolate.
```

Comment:
Would you mind providing an example use-case in this comment? It's hard to come up with one, as a reviewer.

Reply:
There's an example in the original PR post. This would be for allocations that are held but not ever associated with a `v8::ArrayBuffer`